recipes/cuda-nvcc/meta.yaml

```diff
@@ -0,0 +1,95 @@
+{% set name = "cuda-nvcc" %}
+{% set version = "12.0.76" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_nvcc/{{ platform }}/cuda_nvcc-{{ platform }}-{{ version }}-archive.{{ extension }}
+
+  sha256: 160ca9f8828f53daa4b2066c5361aacb45fa2575885f70c223cda1d11df53d6f  # [linux64]
+  sha256: dc5bd7eac57b32a9645c0c23c6a1ba8a9533e846eeec31414e1aa4230afee2f8  # [ppc64le]
+  sha256: 85c8c643ea2cfa398e1b5ffcd597f5bb3b738526c9e3fb8e39dd909c55345f36  # [aarch64]
+  sha256: f67406f44263f0e4a110a8cf0200bff1d3d0e32917db8a7c9e417a87eb7ca372  # [win]
+  patches:
+    nvcc.profile.patch  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+  run_exports:
+    strong:
+      - {{ pin_compatible('cuda-version', max_pin='x') }}
+      - __glibc >=2.17        # [linux]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  host:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+    - cuda-cudart-dev
+    - cuda-nvcc_{{ target_platform }} =={{ version }}
+  run_constrained:
+    - gcc_impl_{{ target_platform }} >=6,<13        # [linux]
+
+test:
+  requires:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - cuda-cudart-dev
+    - cuda-nvcc_{{ target_platform }} =={{ version }}
+  files:
+    - test.cu
+  commands:
+    - nvcc --version
+    - nvcc --verbose test.cu
+
+outputs:
+  - name: cuda-nvcc
+    files:
+      - bin      # [linux]
+      - include  # [linux]
+      - lib      # [linux]
+      - nvvm     # [linux]
+      - Library\nvvm  # [win]
```

Comment:
Does it make sense for `nvvm` to be a separate package? I'm thinking about downstream packages like `Numba` who don't use `nvcc` but do use `nvvm`.

Regardless, this could be split out later if desired.